api/src/main/java/io/grpc/CallOptions.java

```diff
@@ -156,19 +182,19 @@ public Deadline getDeadline() {
    * fails RPCs without sending them if unable to connect.
    */
   public CallOptions withWaitForReady() {
-    CallOptions newOptions = new CallOptions(this);
-    newOptions.waitForReady = Boolean.TRUE;
-    return newOptions;
+    Builder builder = toBuilder(this);
+    builder.waitForReady = Boolean.TRUE;
+    return builder.build();
   }
 
   /**
    * Disables 'wait for ready' feature for the call.
    * This method should be rarely used because the default is without 'wait for ready'.
    */
   public CallOptions withoutWaitForReady() {
-    CallOptions newOptions = new CallOptions(this);
-    newOptions.waitForReady = Boolean.FALSE;
-    return newOptions;
+    Builder builder = new Builder();
```

Comment:
Why is this not using `toBuilder`? It loses previous values from `this` instance.